### PR TITLE
Feature/dragndrop go scripts

### PR DESCRIPTION
**Pull request by JorxSS**
More friendly script creation:

- You can create an script from the editor and will be created with an specific template.  For that, just click on create component -> script, then in the editor you will see an Combobox, just click it and then you can select the previous ones created or Create new one. After putting the name, just open the GameplaySystem solution and manually you will need to add the new Script created. You will find the .cpp and .h created on the same directory as the other scripts inside the GameplaySystem solution.

- You can drag and drop GO into the script editor component to link GO like Unity, to acces that specific GO info, variables, methods, etc.. For that you will need to use the same example code you will find inside the ExampleScript.

- Start and Awake methods for controlling script initialization.

Do QA to check if Hot Reloading and Script creation works. To check if the script works you will need to press the Play button.

### CODE REVIEW

File: Engine/Module/ModuleTime.cpp [hunk 1]
```diff
@@ -1,7 +1,9 @@
 #include "ModuleTime.h"
-#include "Main/Globals.h"
+
 #include "Main/Application.h"
+#include "Main/Globals.h"
 #include "ModuleEditor.h"
+#include "Module/ModuleScriptManager.h"
 #include "ModuleWindow.h"
 #include "UI/EngineLog.h"
```
Comment (Dartemiss) [inline Engine/Module/ModuleTime.cpp:8]: Oh darling "U" go after "H" alphabeticaly :O

File: Game/Resources/Scripts/TemplateScript.cpp [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+#include "UI/Panel/InspectorSubpanel/PanelComponent.h"
+
+#include "imgui.h"
+
+
+
+TemplateScript* TemplateScriptDLL()
+{
+	TemplateScript *instance = new TemplateScript();
```
Comment (Dartemiss) [inline Game/Resources/Scripts/TemplateScript.cpp:19]: It should be according to style: TemplateScript* instance

**Review by Dartemiss — APPROVED** [review]

File: Engine/Component/ComponentScript.h [hunk 3]
```diff
@@ -16,10 +16,16 @@ class ComponentScript :public Component
 	ComponentScript();
 	ComponentScript(GameObject* owner, std::string& script_name);
```
Comment (JoanStinson) [inline Engine/Component/ComponentScript.h]: If you are not going to modify the string always do `const std::string& string_name`! :)

File: Engine/Component/ComponentScript.h [hunk 4]
```diff
@@ -16,10 +16,16 @@ class ComponentScript :public Component
...
 	~ComponentScript() = default;
+
 	void LoadName(std::string& script_name);
```
Comment (JoanStinson) [inline Engine/Component/ComponentScript.h:30]: Here also

File: Engine/Helper/Config.h [hunk 5]
```diff
@@ -47,9 +47,71 @@ class Config
...
+			return_value = std::vector<T>();
+			for (size_t i = 0; i < current_value.Capacity(); ++i)
+			{
+				return_value.push_back(current_value[i].Get<T>());
+			}
+		}
+	};
+
+	template<>
+	void GetVector(const std::string& name, std::vector<std::string>& return_value, const std::vector<std::string>& opt_value) const
```
Comment (JoanStinson) [inline Engine/Helper/Config.h:95]: Duplicated code. I understand you have a template and specific object template for a different use, but in this case the GetVector for string vectors does exactly the same as the generic template for any type! :)

File: Engine/Helper/Utils.cpp [hunk 6]
```diff
@@ -120,3 +120,22 @@ bool Utils::PatchFileName(char* filename)
 	}
 
 }
+
+std::string Utils::LoadFileContent(const std::string& path) {
+	std::ifstream file(path);
+	return std::string((std::istreambuf_iterator<char>(file)), std::istreambuf_iterator<char>());
+}
+
+void Utils::SaveFileContent(const std::string& source, std::string& destination)
```
Comment (JoanStinson) [inline Engine/Helper/Utils.cpp:129]: `const std::string& destination`

File: Engine/UI/Panel/PanelPopups.cpp [hunk 7]
```diff
@@ -33,4 +34,35 @@ void PanelPopups::Render()
...
+
+void PanelPopups::CreateScript()
+{
+	if (create_script_shown)
+	{
+		create_script_shown = false;
+		ImGui::OpenPopup("Create Script");
+	}
+
+	if (ImGui::BeginPopupModal("Create Script", NULL, ImGuiWindowFlags_AlwaysAutoResize))
```
Comment (JoanStinson) [inline Engine/UI/Panel/PanelPopups.cpp:47]: prefer `nullptr` for `NULL` since C++11

File: Game/Resources/Scripts/TemplateScript.cpp (hunk 2 above)
Comment (JoanStinson) [inline Game/Resources/Scripts/TemplateScript.cpp:19]: Is there gonna be only 1 instance? Maybe then make it `static TemplateScript *instance = new TemplateScript();`, so you don't need to create a new object every single time you call `TemplateScriptDLL()`

**Review by JoanStinson — APPROVED** [review]
Great job @JorxSS and @marcorod94!!! Fully imppressed how you managed to do a workaround with the scripts system.

I approve the pull request, but have leaved some recommendations/changes that you don't need to do, but would be nice if you take a look at them, they take 1min. to change so xDD 

Good work!

**Pull request merged**